src/data/bucket/symbol_bucket.js

```diff
@@ -261,28 +301,40 @@ class SymbolBucket {
     }
 
     createArrays() {
-        this.arrays = util.mapObject(symbolInterfaces, (programInterface) => {
+        this.arrays = util.mapObject(this.symbolInterfaces, (programInterface) => {
             return new ArrayGroup(programInterface, this.layers, this.zoom);
         });
     }
 
     prepare(stacks, icons) {
         this.symbolInstances = [];
 
-        // To reduce the number of labels that jump around when zooming we need
-        // to use a text-size value that is the same for all zoom levels.
-        // This calculates text-size at a high zoom level so that all tiles can
-        // use the same value when calculating anchor positions.
-        this.adjustedTextMaxSize = this.layers[0].getLayoutValue('text-size', {zoom: 18});
-        this.adjustedTextSize = this.layers[0].getLayoutValue('text-size', {zoom: this.zoom + 1});
-        this.adjustedIconMaxSize = this.layers[0].getLayoutValue('icon-size', {zoom: 18});
-        this.adjustedIconSize = this.layers[0].getLayoutValue('icon-size', {zoom: this.zoom + 1});
-
         const tileSize = 512 * this.overscaling;
         this.tilePixelRatio = EXTENT / tileSize;
         this.compareText = {};
         this.iconsNeedLinear = false;
 
+        const layer = this.layers[0];
+        if (
+            layer.isLayoutValueFeatureConstant('text-size') &&
+            !layer.isLayoutValueZoomConstant('text-size')
+        ) {
+            this.adjustedTextSize = layer.getLayoutValue('text-size', {zoom: this.zoom + 1});
+        }
+        if (!layer.isLayoutValueZoomConstant('text-size')) {
+            this.textSizeCoveringZoomStops = this.layers[0].getLayoutValueCoveringZoomStops('text-size', this.zoom, this.zoom + 1);
+        }
+
+        if (
+            layer.isLayoutValueFeatureConstant('icon-size') &&
+            !layer.isLayoutValueZoomConstant('icon-size')
+        ) {
+            this.adjustedIconSize = layer.getLayoutValue('icon-size', {zoom: this.zoom + 1});
+        }
+        if (!layer.isLayoutValueZoomConstant('icon-size')) {
+            this.iconSizeCoveringZoomStops = this.layers[0].getLayoutValueCoveringZoomStops('icon-size', this.zoom, this.zoom + 1);
+        }
```

Comment:
Hmm.  This is overcomplicated.  We don't really need to serialize `adjusted{Text,Icon}Size`, or `{text,icon}SizeCoveringZoomStops` on the bucket -- drawSymbols() can (re)compute these at render time, since they only depend on the tile's zoom level.